examples/example-gateway/clients/baz/baz.go

```diff
@@ -0,0 +1,237 @@
+func NewClient(config *zanzibar.StaticConfig, gateway *zanzibar.Gateway) *BazClient {
+	// TODO: (lu) get clientName from config
+	// clientName := config.MustGetString("clients.baz.clientName")
+	clientName := "BazClient"
+
+	// TODO: (lu) client channel opts
+	ch, err := tchannel.NewChannel(clientName, nil)
+	if err != nil {
+		gateway.Logger.Error("Failed to create tchannel",
+			zap.String("error", err.Error()))
+	}
+
+	ip := config.MustGetString("clients.SimpleService.ip")
+	port := config.MustGetInt("clients.SimpleService.port")
+	ch.Peers().Add(ip + ":" + strconv.Itoa(int(port)))
+
+	// TODO: (lu) get serviceName from config
+	// serviceName := config.MustGetString("clients.baz.serviceName")
+	serviceName := "SimpleService"
+	client := zt.NewClient(ch, serviceName)
+
+	return &BazClient{
+		SimpleServiceClient{
+			thriftService: serviceName,
+			client:        client,
+		},
+	}
+}
+
+// SimpleServiceClient is the client to talk to SimpleService backend.
+type SimpleServiceClient struct {
+	thriftService string
+	client        zt.TChanClient
+}
+
+// Call ...
+func (c *SimpleServiceClient) Call(ctx context.Context, reqHeaders map[string]string, args *baz.SimpleService_Call_Args) (map[string]string, *baz.BazResponse, error) {
+	var result baz.SimpleService_Call_Result
+	respHeaders, success, err := c.client.Call(ctx, c.thriftService, "Call", reqHeaders, args, &result)
+	if err == nil && !success {
+		err = errors.New("received no result or unknown exception for Call")
+	}
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := baz.SimpleService_Call_Helper.UnwrapResponse(&result)
+
+	return respHeaders, resp, err
+}
+
+// Simple ...
+func (c *SimpleServiceClient) Simple(ctx context.Context, reqHeaders map[string]string) (map[string]string, error) {
+	var result baz.SimpleService_Simple_Result
+	args := baz.SimpleService_Simple_Args{}
+	respHeaders, success, err := c.client.Call(ctx, c.thriftService, "Simple", reqHeaders, &args, &result)
+	if err == nil && !success {
+		switch {
+		case result.SimpleErr != nil:
+			err = result.SimpleErr
+		default:
+			err = errors.New("received no result or unknown exception for Simple")
+		}
+	}
+
+	return respHeaders, err
+}
+
+// SimpleFuture ...
+func (c *SimpleServiceClient) SimpleFuture(ctx context.Context, reqHeaders map[string]string) (map[string]string, error) {
+	var result baz.SimpleService_SimpleFuture_Result
+	args := baz.SimpleService_SimpleFuture_Args{}
+	respHeaders, success, err := c.client.Call(ctx, c.thriftService, "SimpleFuture", reqHeaders, &args, &result)
+	if err == nil && !success {
+		switch {
+		case result.SimpleErr != nil:
+			err = result.SimpleErr
+		case result.NewErr != nil:
+			err = result.NewErr
+		default:
+			err = errors.New("received no result or unknown exception for SimpleFuture")
+		}
+	}
+
+	return respHeaders, err
+}
+
+// SimpleServiceServer is the Baz backend service
+type SimpleServiceServer struct {
```

Comment:
Who uses `SimpleServiceServer` ? It looks like this code doesn't belong here.

Can we move it to `endpoints/baz/baz.go` or something ?